mica/stats/guide_stats.py

```diff
@@ -0,0 +1,552 @@
+    return obsids
+
+
+def calc_stats(obsid):
+    obspar = mica.archive.obspar.get_obspar(obsid)
+    if not obspar:
+        raise ValueError("No obspar for {}".format(obsid))
+    manvr = None
+    dwell = None
+    try:
+        manvrs = events.manvrs.filter(obsid=obsid, n_dwell__gt=0)
+        dwells = events.dwells.filter(obsid=obsid)
+        if dwells.count() == 1 and manvrs.count() == 0:
+            # If there is more than one dwell for the manvr but they have
+            # different obsids (unusual) so don't throw an overlapping interval kadi error
+            # just get the maneuver to the attitude with this dwell
+            dwell = dwells[0]
+            manvr = dwell.manvr
+        elif dwells.count() == 0:
+            # If there's just nothing, that doesn't need an error here
+            # and gets caught outside the try/except
+            pass
+        else:
+            # Else just take the first matches from each
+            manvr = manvrs[0]
+            dwell = dwells[0]
+    except ValueError:
+        multi_manvr = events.manvrs.filter(start=obspar['tstart'] - 100000,
+                                           stop=obspar['tstart'] + 100000)
+        multi = multi_manvr.select_overlapping(events.obsids(obsid=obsid))
+        deltas = [np.abs(m.tstart - obspar['tstart']) for m in multi]
+        manvr = multi[np.argmin(deltas)]
+        dwell = manvr.dwell_set.first()
+
+    if not manvr or not dwell:
+        raise ValueError("No manvr or dwell for {}".format(obsid))
+    if not manvr.get_next():
+        raise ValueError("No *next* manvr so can't calculate dwell".format(obsid))
+    if not manvr.guide_start:
+        raise ValueError("No guide transition for {}".format(obsid))
+    if not manvr.kalman_start:
+        raise ValueError("No Kalman transition for {}".format(obsid))
+
+
+    logger.info("Found obsid manvr at {}".format(manvr.start))
+    logger.info("Found dwell at {}".format(dwell.start))
+    guide_start = manvr.guide_start
+    starcheck = get_starcheck_catalog_at_date(manvr.guide_start)
+    if starcheck is None or 'cat' not in starcheck or not len(starcheck['cat']):
+        raise ValueError('No starcheck catalog found for {}'.format(
+                manvr.get_obsid()))
+    starcat_time = DateTime(starcheck['cat']['mp_starcat_time'][0]).secs
+    starcat_dtime = starcat_time - DateTime(manvr.start).secs
+    # If it looks like the wrong starcheck by time, give up
+    if abs(starcat_dtime) > 300:
+        raise ValueError("Starcheck cat time delta is {}".format(starcat_dtime))
+    if abs(starcat_dtime) > 30:
+        logger.warn("Starcheck cat time delta of {} is > 30 sec".format(abs(starcat_dtime)))
+    vals, times, star_info = get_data(start=manvr.stop, stop=manvr.get_next().start,
+                                      obsid=obsid, starcheck=starcheck)
+    gui_stats = calc_gui_stats(manvr.stop, manvr.get_next().start, vals, times, star_info)
```

Comment:
Just Kalman.  We are more interested in mean effects than edge effects near the beginning of an observation.